docs/rules/component-definition-name-casing.md

```diff
@@ -0,0 +1,47 @@
+# enforce specific casing for component definition name (vue/component-definition-name-casing)
+
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+Define a style for component definition name casing for consistency purposes.
+
+## :book: Rule Details
+
+:+1: Examples of **correct** code for `PascalCase`:
+
+```js
+export default {
+  name: 'MyComponent'
+}
+```
+```js
+Vue.component('MyComponent', {
+  
+})
+```
+
```

Comment:
@armano2 Hello!
Can this rule be verified if the user defines only the name of the component as follows?

```js
const component = {} // ex. import external components
Vue.component('MyComponent', component)
```

Reply:
Its working like that, but i will add additional test for it

Reply:
@ota-meshi i did a little more testing, and there was few uncovered cases, now it should be ok :>